app/components/documents/document-card.js

```diff
@@ -64,6 +66,15 @@ export default class DocumentsDocumentCardComponent extends Component {
       && this.currentSession.may('manage-signatures');
   }
 
+  @task
+  *loadPieceMayBeEditedOrDeleted() {
+    const signCompletionActivity = yield this.piece.signCompletionActivity;
+    const signedPiece = yield this.piece.signedPiece;
+    if (!this.signMarkingActivity && !signCompletionActivity && !signedPiece){
+      this.mayEditOrDelete = true;
+    }
+  }
+
```

Comment:
This could fall under `loadSignatureRelatedData`
Could also be util/service call (loading `signMarkingActivity` is ok, since the second await/yield will be resolved instantly)